ledger/applications.go

```diff
@@ -235,3 +237,11 @@ func (al *logicLedger) DelGlobal(key string) error {
 func (al *logicLedger) GetDelta(txn *transactions.Transaction) (evalDelta basics.EvalDelta, err error) {
 	return al.cow.BuildEvalDelta(al.aidx, txn)
 }
+
+func (al *logicLedger) AppendLog(txn *transactions.Transaction, value string) error {
+	idx, err := txn.IndexByAppID(txn.ApplicationID)
+	if err != nil {
+		return err
+	}
```

Comment:
Just to be paranoid, could you return an error if idx != 0? It shouldn't happen, but it would be really bad if it did happen